### PR TITLE
Changed some method signatures to reduce code clutter.

### PR DESCRIPTION
**Pull request by twalmsley**
This helps to make the client code cleaner and clearer.

### CODE REVIEW

File: hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java [hunk 1]
```diff
@@ -55,10 +55,18 @@ public interface Top {
     /**
      * Get predicate value(s) by predicate Object.
      *
-     * @param predicateId Predicate ID.
+     * @param predicateId Predicate ID. {@link IRI}
```
Comment (GCHQDeveloper42) [inline hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java:58]:
```suggestion
     * @param predicateIri Predicate {@link IRI}.
```

File: hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java [hunk 2]
```diff
@@ -55,10 +55,18 @@ public interface Top {
...
-    Set<Object> value(Object predicateId);
+    <T> Set<T> values(IRI predicateId);
+
+    /**
+     * Get predicate value(s) by predicate Object.
+     *
+     * @param predicateId Predicate ID. {@link IRI}
+     * @return A single value of type T or null (Object or string-literals).
+     */
+    <T> T oneValue(IRI predicateId);
```
Comment (GCHQDeveloper42) [inline hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java:69]:
```suggestion
     * @param predicateId Predicate {@link IRI}.
     * @return A single value of type T or null (Object or string-literals).
     */
    <T> T oneValue(IRI predicateIri);
```

File: hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java [hunk 3]
```diff
@@ -55,10 +55,18 @@ public interface Top {
     /**
      * Get predicate value(s) by predicate Object.
      *
-     * @param predicateId Predicate ID.
+     * @param predicateId Predicate ID. {@link IRI}
      * @return Set of predicate values (Object or string-literals).
      */
-    Set<Object> value(Object predicateId);
+    <T> Set<T> values(IRI predicateId);
```
Comment (GCHQDeveloper42) [inline hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java:61]:
```suggestion
    <T> Set<T> values(IRI predicateIri);
```

File: hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java (hunk 1 above)
Comment (twalmsley) [reply hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java:58]: I think these changes have gone in on the `191-thingaddvalue-is-not-tight-enough` branch as part of a separate task.

File: hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java (hunk 2 above)
Comment (twalmsley) [reply hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java]: Resolved in new commit.

File: hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java (hunk 3 above)
Comment (twalmsley) [reply hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java]: Resolved in new commit.

File: hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java (hunk 1 above)
Comment (twalmsley) [reply hqdm/src/main/java/uk/gov/gchq/magmacore/hqdm/pojo/Top.java]: Resolved in new commit.

**Review by GCHQDeveloper42 — APPROVED** [review]

**Pull request merged**